packages/rulesets/src/oas/index.ts

```diff
@@ -671,12 +710,7 @@ const ruleset = {
       formats: [oas3],
       recommended: true,
       type: 'validation',
-      given: [
-        "$.components.schemas..[?(@property !== 'properties' && @ && (@ && @.example !== void 0 || @.default !== void 0) && (@.enum || @.type || @.format || @.$ref || @.properties || @.items))]",
-        "$..content..[?(@property !== 'properties' && @ && (@ && @.example !== void 0 || @.default !== void 0) && (@.enum || @.type || @.format || @.$ref || @.properties || @.items))]",
-        "$..headers..[?(@property !== 'properties' && @ && (@ && @.example !== void 0 || @.default !== void 0) && (@.enum || @.type || @.format || @.$ref || @.properties || @.items))]",
-        "$..parameters..[?(@property !== 'properties' && @ && (@ && @.example !== void 0 || @.default !== void 0) && (@.enum || @.type || @.format || @.$ref || @.properties || @.items))]",
-      ],
+      given: ['#SchemaObject', '#SchemaObject..[?(@ && @.example)]'],
```

Comment:
Won't this capture, `components.schema.MySchema.properties` if there's a property called `example`? Or maybe I'm missing something?

Reply:
It will, although the current impl does it as well.
In any case, before I mark that PR as ready to review, I'll need to figure out a way to do it. It's not that trivial using  JSONPath expressions.,so I'll probably need to think of something else

Reply:
Apologies, I totally missed that this was a draft!